src/Fixer/Casing/LowercaseStaticReferenceFixer.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\Casing;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author Kuba Werłos <werlos@gmail.com>
+ */
+final class LowercaseStaticReferenceFixer extends AbstractFixer
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'Class static references `self`, `static` and `parent` MUST be in lower case.',
+            [new CodeSample('<?php
+class Foo extends Bar
+{
+    public function baz()
+    {
+        return 42;
+    }
+
+    public function qux()
+    {
+        return SELF::baz();
+    }
+
+    public function quux()
+    {
+        return STATIC::baz();
+    }
+
+    public function quuz()
+    {
+        return PARENT::baz();
+    }
+}
+')]
+        );
+    }
+
+    public function isCandidate(Tokens $tokens)
+    {
+        return $tokens->isAnyTokenKindsFound([T_STATIC, T_STRING]);
+    }
+
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        for ($index = $tokens->count() - 1; $index >= 0; --$index) {
+            $token = $tokens[$index];
+
+            if (!$token->isGivenKind([T_STATIC, T_STRING])) {
+                continue;
+            }
+
+            $nextIndex = $tokens->getNextMeaningfulToken($index);
+            if (!$tokens[$nextIndex]->isGivenKind([T_PAAMAYIM_NEKUDOTAYIM])) {
+                continue;
```

Comment:
What about supporting `instanceof` as well, e.g. `$foo instanceof self`? And typehints?

Reply:
@julienfalque done, ready for review.